homeassistant/components/websocket_api/commands.py

```diff
@@ -420,3 +421,23 @@ async def handle_test_condition(hass, connection, msg):
     connection.send_result(
         msg["id"], {"result": check_condition(hass, msg.get("variables"))}
     )
+
+
+@decorators.websocket_command(
+    {
+        vol.Required("type"): "execute_script",
+        vol.Required("sequence"): cv.SCRIPT_SCHEMA,
+    }
+)
+@decorators.require_admin
+@decorators.async_response
+async def handle_execute_script(hass, connection, msg):
+    """Handle execute script command."""
+    # Circular dep
+    # pylint: disable=import-outside-toplevel
+    from homeassistant.helpers.script import Script
+
+    context = connection.context(msg)
+    script_obj = Script(hass, msg["sequence"], f"{const.DOMAIN} script", const.DOMAIN)
+    await script_obj.async_run(context=context)
```

Comment:
One thing I am realizing now is that we don't currently cancel ongoing tasks when the websocket connection closes. I wonder if we should? Or maybe we push that out to the future.

Reply:
Let's do that in another PR 👍